kolibri/core/templatetags/core_tags.py

```diff
@@ -49,3 +52,22 @@ def frontend_base_head_markup():
     :return: HTML to insert into head of base.html
     """
     return FrontEndBaseHeadHook.html()
+
+
+@register.simple_tag()
+def theme_favicon():
+    """
+    Render a favicon link to put in the <head> tag of base.html, if a favicon is
+    provided by the theme. If not, a default will be returned.
+    """
+    favicon_urls = [
+        logo["src"]
+        for logo in ThemeHook.get_theme().get("logos", [])
+        if logo.get("content_type", "") == "image/vnd.microsoft.icon"
+    ]
+
+    # Choose the first available .ico file. It's unlikely there's more than
+    # one specified in the theme.
+    favicon_url = favicon_urls[0] if favicon_urls else static("logo.ico")
```

Comment:
Because you have moved the `logo.ico` into the default_theme plugin, this will now fail if there is another theme active, because default_theme will not be registered as an INSTALLED_APP for Django. Probably easiest to leave the ico file where it is currently and just reference it from `assets/logo.ico`.

Reply:
Sorry, I’m still learning all the things to watch out for with Django :slightly_smiling_face: Fixed!